tee-worker/enclave-runtime/src/rpc/worker_api_direct.rs

```diff
@@ -304,6 +258,56 @@ where
 				Err(_) => Ok(json!(compute_hex_encoded_return_error("parse error"))),
 			}
 		});
+
+		// state_getStorage
+		let state_get_storage = "state_getStorage";
+		io.add_sync_method(state_get_storage, move |params: Params| {
+			if state_storage.is_none() {
+				return Ok(json!(compute_hex_encoded_return_error(
+					"state_getStorage is not avaiable"
+				)))
+			}
+			let state_storage = state_storage.clone().unwrap();
+			match params.parse::<(String, String)>() {
+				Ok((shard_str, key_hash)) => {
+					let key_hash = if key_hash.starts_with("0x") {
+						key_hash.strip_prefix("0x").unwrap()
+					} else {
+						key_hash.as_str()
+					};
+					let key_hash = match hex::decode(key_hash) {
+						Ok(key_hash) => key_hash,
+						Err(_) =>
+							return Ok(json!(compute_hex_encoded_return_error("docode key error"))),
+					};
+
+					let shard: ShardIdentifier = match decode_shard_from_base58(shard_str.as_str())
+					{
+						Ok(id) => id,
+						Err(msg) => {
+							let error_msg = format!("decode shard failure due to: {}", msg);
+							return Ok(json!(compute_hex_encoded_return_error(error_msg.as_str())))
+						},
+					};
+					match state_storage.load_cloned(&shard) {
+						Ok((state_storage, _hash)) => {
+							// Get storage by key hash
+							let value =
+								state_storage.get(key_hash.as_slice()).cloned().unwrap_or_default();
+							debug!("query storage value:{:?}", &value);
+							let json_value =
+								RpcReturnValue::new(value, false, DirectRequestStatus::Ok);
+							Ok(json!(json_value.to_hex()))
+						},
+						Err(e) => {
+							let error_msg = format!("load shard failure due to: {:?}", e);
+							return Ok(json!(compute_hex_encoded_return_error(error_msg.as_str())))
+						},
+					}
+				},
+				Err(_err) => Ok(json!(compute_hex_encoded_return_error("parse error"))),
+			}
+		});
```

Comment:
Is it only switch the code position?

Reply:
Yes, it's moved inside condition `if cfg!(not(feature = "production"))` so this rpc method won't be present with `production` feature on.